components/brave_wallet/browser/brave_wallet_service.h

```diff
@@ -6,39 +6,90 @@
 #include "base/scoped_observer.h"
+#include "base/values.h"
 #include "components/keyed_service/core/keyed_service.h"
+#include "extensions/browser/extension_registry.h"
+#include "extensions/browser/extension_registry_observer.h"
+#include "url/gurl.h"
+
+class BrowserWalletDelegate;
+class PrefChangeRegistrar;
+class PrefService;
 
 namespace base {
+class FilePath;
 class SequencedTaskRunner;
 }  // namespace base
 
 namespace content {
 class BrowserContext;
 }  // namespace content
 
-class BraveWalletController;
-class Profile;
-
-class BraveWalletService : public KeyedService {
+class BraveWalletService : public KeyedService,
+    public extensions::ExtensionRegistryObserver {
  public:
-  explicit BraveWalletService(content::BrowserContext* context);
+  explicit BraveWalletService(content::BrowserContext* context,
+      std::unique_ptr<BrowserWalletDelegate> browser_wallet_delegate);
   ~BraveWalletService() override;
 
   bool Init();
-  BraveWalletController* controller() const { return controller_.get(); }
+
+  void ResetCryptoWallets(const base::FilePath& profile_path);
+  void CloseTabsAndRestart();
+  bool LoadRootSeedInfo(std::vector<uint8_t> key, std::string* seed);
+
+  std::string GetWalletSeed(std::vector<uint8_t> key);
+  std::string GetBitGoSeed(std::vector<uint8_t> key);
+
+  static std::string GetEthereumRemoteClientSeedFromRootSeed(
+      const std::string& seed);
+  static std::string GetBitGoSeedFromRootSeed(
+      const std::string& seed);
+  static bool SealSeed(const std::string& seed, const std::string& key,
+      const std::string& nonce, std::string* cipher_seed);
+  static bool OpenSeed(const std::string& cipher_seed,
+      const std::string& key, const std::string& nonce, std::string* seed);
+  static void SaveToPrefs(PrefService* prefs, const std::string& cipher_seed,
+      const std::string& nonce);
+  static bool LoadFromPrefs(PrefService* prefs, std::string* cipher_seed,
+      std::string* nonce);
+  static std::string GetRandomNonce();
+  static std::string GetRandomSeed();
+  static const size_t kNonceByteLength;
+  static const size_t kSeedByteLength;
+
+ protected:
+  void RemoveUnusedWeb3ProviderContentScripts();
+  void OnPreferenceChanged();
+  // ExtensionRegistryObserver implementation.
+  void OnExtensionLoaded(content::BrowserContext* browser_context,
+                         const extensions::Extension* extension) override;
+  void OnCryptoWalletsReset(bool success);
 
  private:
   content::BrowserContext* context_;
-  std::unique_ptr<BraveWalletController> controller_;
-  base::WeakPtrFactory<BraveWalletService> weak_factory_;
+  std::unique_ptr<PrefChangeRegistrar> pref_change_registrar_;
+  std::unique_ptr<BrowserWalletDelegate> browser_wallet_delegate_;
+  ScopedObserver<extensions::ExtensionRegistry,
+      extensions::ExtensionRegistryObserver> extension_registry_observer_{this};
+  scoped_refptr<base::SequencedTaskRunner> file_task_runner_;
 
+ private:
```

Comment:
you have an extra `private` here

Reply:
fixed, from when merging in the controller